src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -0,0 +1,774 @@
+        {
+            if (this._rows.Count > 0)
+            {
+                this._state = State.ProcessingChanges;
+                IReadOnlyList<SqlChange<T>> changes = null;
+
+                try
+                {
+                    // What should we do if this fails? It doesn't make sense to retry since it's not a connection based
+                    // thing. We could still try to trigger on the correctly processed changes, but that adds additional
+                    // complication because we don't want to release the leases on the incorrectly processed changes.
+                    // For now, just give up I guess?
+                    changes = this.GetChanges();
+                }
+                catch (Exception e)
+                {
+                    await this.ClearRowsAsync(
+                        $"Failed to extract user table data from table {this._userTable.FullName} associated " +
+                        $"with change metadata due to error: {e.Message}", true);
+                }
+
+                if (changes != null)
+                {
+                    FunctionResult result = await this._executor.TryExecuteAsync(
+                        new TriggeredFunctionData() { TriggerValue = changes },
+                        this._cancellationTokenSourceExecutor.Token);
+
+                    if (result.Succeeded)
+                    {
+                        await this.ReleaseLeasesAsync(token);
+                    }
+                    else
+                    {
+                        // In the future might make sense to retry executing the function, but for now we just let
+                        // another worker try.
+                        await this.ClearRowsAsync(
+                            $"Failed to trigger user's function for table {this._userTable.FullName} due to " +
+                            $"error: {result.Exception.Message}", true);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Executed once every <see cref="LeaseTime"/> period. If the state of the change monitor is
+        /// <see cref="State.ProcessingChanges"/>, then we will renew the leases held by the change monitor on "_rows".
+        /// </summary>
+        private async void RunLeaseRenewalLoopAsync()
+        {
+            try
+            {
+                CancellationToken token = this._cancellationTokenSourceRenewLeases.Token;
+
+                using (var connection = new SqlConnection(this._connectionString))
+                {
+                    await connection.OpenAsync(token);
+
+                    while (!token.IsCancellationRequested)
+                    {
+                        await this._rowsLock.WaitAsync(token);
+
+                        await this.RenewLeasesAsync(connection, token);
+
+                        // Want to make sure to renew the leases before they expire, so we renew them twice per lease period.
```

Comment:
I'm not sure just twice per period is enough - it's essentially just 1 since the other will be happening as the lease is expiring so could result in another worker coming in between when the lease expires and the 2nd renewal happens (in case the first failed for some reason).

At least 4 per period would be a lot safer.

Reply:
Thought the comment says otherwise, the code intends to have enough time to get the lease renewed once only before it expired. It assumes that it will always succeed if the worker is up and running. I will update the `LeaseIntervalInSeconds` to 60 seconds and renewal interval to 15 seconds.

Reply:
Done. @Charles-Gagnon, please review the updated renewal intervals.

Reply:
Looks better, thanks. Could consider making it user-configurable later on to give users more direct control over the performance but this should be fine for now.